ultravox/data/datasets.py

```diff
@@ -274,10 +275,15 @@ def __init__(self, args: VoiceDatasetArgs) -> None:
         self._args = args
         self._session: Optional[requests.Session] = None
         self._rng = np.random.default_rng(self._args.shuffle_seed)
+        self._weight = 1.0  # the default weight for the dataset
```

Comment:
Hmm, having weight inside dataset seems a bit weird.

Reply:
Each dataset should have a weight (so that we can used weighted sampling), and the default is 1.0. In the GenericVoiceDataset, the default weight is 1.0 and can be configured in the config file. Currently, all other dataset classes share the same weight because these classes don’t support configurations. Once we update all dataset classes to support configuration via the YAML config, their weight can be configured individually (with the default weight being 1.0).

Reply:
FWIW, `datasets.interleave_datasets` passes in `probabilities` as an array of probabilities, allowing this logic to stay within the interleave method. I think that might turn out to be cleaner here, although it might require a bit more work to set up the config.